components/layout/construct.rs

```diff
@@ -1248,9 +1255,15 @@ impl<'a, ConcreteThreadSafeLayoutNode: ThreadSafeLayoutNode>
             // Go to a path that concatenates our kids' fragments.
             self.build_fragments_for_nonreplaced_inline_content(node)
         } else {
-            // Otherwise, just nuke our kids' fragments, create our fragment if any, and be done
-            // with it.
-            self.build_fragments_for_replaced_inline_content(node)
+            if node.type_id() == Some(LayoutNodeType::Element(LayoutElementType::HTMLMediaElement))
+            {
+                // Do not treat media elements as leafs.
+                self.build_flow_for_block(node, None)
```

Comment:
This is wrong. `data:text/html,a <video controls style="display: inline"></video> b` should be in the same line, so `<video>` shouldn't be a block.

Reply:
I ended up reverting this change. I couldn't make it work for inline layout :(. I'll file a follow up to fix this.